### Problem Statement

D'Paragon owns a supermarket chain across the country. Each major branch is located in 3 cities - Lagos, Abuja and PH. You have been hired as a data analyst to help them understand sales trend and also measure growth to be able compete with other supermarkets in the same location. 


- CRISP - DM framework 

1. Data Understanding / Exploration

In [1]:
# data structure libraries
import numpy as np
import pandas as pd

# visualization libraries
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

C:\Users\aaginam\AppData\Local\anaconda3\Lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


In [2]:
# read in the data

lagos_df =  pd.read_csv('Datasets/Lagos_Branch.csv') 
abuja_df =  pd.read_csv('Datasets/Abuja_Branch.csv')
ph_df =  pd.read_csv('Datasets/Port_Harcourt_Branch.csv')

In [4]:
lagos_df.tail()

Invoice ID Branch   City Customer type  Gender            Product line  \
335  886-18-2897      A  Lagos        Normal  Female      Food and beverages   
336  745-74-0715      A  Lagos        Normal    Male  Electronic accessories   
337  727-02-1313      A  Lagos        Member    Male      Food and beverages   
338  347-56-2442      A  Lagos        Normal    Male      Home and lifestyle   
339  849-09-3807      A  Lagos        Member  Female     Fashion accessories   

     Unit price  Quantity    Tax 5%      Total       Date   Time Payment  \
335     20361.6         5   5090.40  106898.40  3/22/2019  19:06    Card   
336     20890.8         2   2089.08   43870.68  3/10/2019  20:46    Epay   
337     11462.4         1    573.12   12035.52   2/9/2019  13:22    Cash   
338     23695.2         1   1184.76   24879.96  2/22/2019  15:33    Cash   
339     31802.4         7  11130.84  233747.64  2/18/2019  13:28    Cash   

         cogs  gross margin percentage  gross income  Rating  
335  101808.0                 4.761905       5090.40     4.5  
336   41781.6                 4.761905       2089.08     8.8  
337   11462.4                 4.761905        573.12     7.7  
338   23695.2                 4.761905       1184.76     4.1  
339  222616.8                 4.761905      11130.84     6.6

In [5]:
ph_df.head()

Invoice ID Branch           City Customer type  Gender  \
0  226-31-3081      C  Port Harcourt        Normal  Female   
1  699-14-3026      C  Port Harcourt        Normal    Male   
2  315-22-5665      C  Port Harcourt        Normal  Female   
3  300-71-4605      C  Port Harcourt        Member    Male   
4  183-56-6882      C  Port Harcourt        Member  Female   

             Product line  Unit price  Quantity    Tax 5%      Total  \
0  Electronic accessories      5500.8         5   1375.20   28879.20   
1  Electronic accessories     30740.4         7  10759.14  225941.94   
2      Home and lifestyle     26481.6        10  13240.80  278056.80   
3  Electronic accessories     30974.4         5   7743.60  162615.60   
4      Food and beverages     35791.2         4   7158.24  150323.04   

        Date   Time Payment      cogs  gross margin percentage  gross income  \
0   3/8/2019  10:29    Cash   27504.0                 4.761905       1375.20   
1  3/25/2019  18:30    Epay  215182.8                 4.761905      10759.14   
2  2/24/2019  11:38    Epay  264816.0                 4.761905      13240.80   
3  2/25/2019  11:24    Epay  154872.0                 4.761905       7743.60   
4   2/6/2019  10:42    Epay  143164.8                 4.761905       7158.24   

   Rating  
0     9.6  
1     4.1  
2     8.0  
3     4.8  
4     7.5

In [6]:
abuja_df.head()

Invoice ID Branch   City Customer type  Gender            Product line  \
0  692-92-5582      B  Abuja        Member  Female      Food and beverages   
1  351-62-0822      B  Abuja        Member  Female     Fashion accessories   
2  529-56-3974      B  Abuja        Member    Male  Electronic accessories   
3  299-46-1805      B  Abuja        Member  Female       Sports and travel   
4  319-50-3348      B  Abuja        Normal  Female      Home and lifestyle   

   Unit price  Quantity    Tax 5%      Total       Date   Time Payment  \
0     19742.4         3   2961.36   62188.56  2/20/2019  13:27    Card   
1      5212.8         4   1042.56   21893.76   2/6/2019  18:07    Epay   
2      9183.6         4   1836.72   38571.12   3/9/2019  17:03    Cash   
3     33739.2         6  10121.76  212556.96  1/15/2019  16:19    Cash   
4     14508.0         2   1450.80   30466.80  3/11/2019  15:30    Epay   

       cogs  gross margin percentage  gross income  Rating  
0   59227.2                 4.761905       2961.36     5.9  
1   20851.2                 4.761905       1042.56     4.5  
2   36734.4                 4.761905       1836.72     6.8  
3  202435.2                 4.761905      10121.76     4.5  
4   29016.0                 4.761905       1450.80     4.4

In [7]:
comb_df = pd.concat([lagos_df, abuja_df, ph_df], ignore_index = True)

In [9]:
comb_df.tail()

Invoice ID Branch           City Customer type  Gender  \
995  148-41-7930      C  Port Harcourt        Normal    Male   
996  189-40-5216      C  Port Harcourt        Normal    Male   
997  267-62-7380      C  Port Harcourt        Member    Male   
998  652-49-6720      C  Port Harcourt        Member  Female   
999  233-67-5758      C  Port Harcourt        Normal    Male   

               Product line  Unit price  Quantity    Tax 5%      Total  \
995       Health and beauty     35985.6         7  12594.96  264494.16   
996  Electronic accessories     34693.2         7  12142.62  254995.02   
997  Electronic accessories     29642.4        10  14821.20  311245.20   
998  Electronic accessories     21942.0         1   1097.10   23039.10   
999       Health and beauty     14526.0         1    726.30   15252.30   

          Date   Time Payment      cogs  gross margin percentage  \
995  1/23/2019  10:33    Cash  251899.2                 4.761905   
996   1/9/2019  11:40    Cash  242852.4                 4.761905   
997  3/29/2019  19:12    Epay  296424.0                 4.761905   
998  2/18/2019  11:40    Epay   21942.0                 4.761905   
999  1/29/2019  13:46    Epay   14526.0                 4.761905   

     gross income  Rating  
995      12594.96     6.1  
996      12142.62     6.0  
997      14821.20     4.3  
998       1097.10     5.9  
999        726.30     6.2

In [12]:
# PH
comb_df.loc[comb_df.Branch == 'C'].shape

(328, 17)

In [13]:
# Lagos
comb_df[comb_df['Branch'] == 'A'].shape

(340, 17)

In [14]:
# Abuja
comb_df[comb_df['Branch'] == 'B'].shape

(332, 17)

In [15]:
comb_df.shape

(1000, 17)

The no of rows of data for each branch is 

- Lagos - 340
- Abuja - 332
- PH - 328

In [16]:
comb_df.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Total', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

In [17]:
comb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

#### Todo

1. Combine the date and time column into a single column 

2. Cast the column to a datetime column

3. Rename Payment to Payment Type 